controller/internal/controller/consumer_controller.go

```diff
@@ -181,3 +202,70 @@ func (r *ConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		)
 	return controller.Complete(r)
 }
+
+// checkConsumerConflicts Perform full conflict detection and rebuild the index
+func (r *ConsumerReconciler) checkConsumerConflicts(ctx context.Context, state *consumerReconcileState) error {
+	// Clear old indexes
+	r.keyIndex.index = make(map[string]map[string]map[string]string)
+
+	// Check conflicts for all valid consumers in the current state
+	for ns, consumers := range state.namespaceToConsumers {
+		for _, consumer := range consumers {
+			if err := r.indexConsumer(ns, consumer); err != nil {
+				log.Error("Conflict detected")
+				consumer.SetAccepted(mosniov1.ReasonInvalid, err.Error())
+			}
+		}
+	}
+	return nil
+}
+
+// indexConsumer method to use the plugin's Index method
+func (r *ConsumerReconciler) indexConsumer(namespace string, consumer *mosniov1.Consumer) error {
+	r.keyIndex.mu.Lock()
+	defer r.keyIndex.mu.Unlock()
+
+	if consumer == nil || consumer.Spec.Auth == nil {
+		return fmt.Errorf("nil consumer or auth config")
+	}
+
+	for pluginName, plugin := range consumer.Spec.Auth {
+
+		p, ok := plugins.LoadPlugin(pluginName).(plugins.Plugin)
```

Comment:
Better to use plugin.ConsumerPlugin so we don't need to define another Indexer interface.